core/src/main/java/org/apache/iceberg/view/ViewMetadata.java

```diff
@@ -283,20 +292,25 @@ private int addVersionInternal(ViewVersion version) {
         }
       }
 
-      ViewVersion newVersion;
       if (newVersionId != version.versionId()) {
-        newVersion = ImmutableViewVersion.builder().from(version).versionId(newVersionId).build();
+        version = ImmutableViewVersion.builder().from(version).versionId(newVersionId).build();
+      }
```

Comment:
Nit: Now that the version may be updated more than once, it probably makes more sense to relocate this to just after the new version ID is assigned.

Reply:
done